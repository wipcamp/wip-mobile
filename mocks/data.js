export const newProblem = {
    topic: "Test new problem",
    problem_type_id: 1,
    description: "Test new problem",
    report_id: 1
}

export const problem = [
    {
        id: 1,
        topic: "Test Problem 1",
        description: "Test Description 1",
        problem_type_id: 1,
        report_id: 1,
        is_solve: false,
        not_solve: false,
        created_at: "2018-02-02 17:50:01",
        updated_at: "2018-02-02 17:50:01"
    },
    {
        id: 2,
        topic: "Test Problem 2",
        description: "Test Description 2",
        problem_type_id: 2,
        report_id: 2,
        is_solve: false,
        not_solve: false,
        created_at: "2018-02-02 17:50:02",
        updated_at: "2018-02-02 17:50:02"
    }
]

export const categoryProblem = [
    {
        value: 1,
        label: "Test 1"
    },
    {
        value: 2,
        label: "Test 2"
    }
]

export const filter = 0

export const filterView = [
    {
        value: 0,
        label: "ทั้งหมด"
    },
    {
        label: 1,
        value: "Test 1"
    },
    {
        label: 2,
        value: "Test 2"
    }
]