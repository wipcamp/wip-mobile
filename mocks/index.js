const mockComponent = require.requireActual('react-native/jest/mockComponent')

module.exports = {
    initAll: () => {
      jest.mock('ScrollView', () => {
        const RealScrollView = require.requireActual('ScrollView')
        const ScrollView = mockComponent('ScrollView')
        ScrollView.prototype.render = RealScrollView.prototype.render
        return ScrollView
      })

      jest.mock('Dimensions', () => {
        const Dimensions = {
          get: jest.fn().mockReturnValue({width: 100, height:100})
        }
        return Dimensions
      })
    }
  }