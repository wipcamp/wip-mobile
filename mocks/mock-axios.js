import axios from 'axios'
import mockAxios from 'axios-mock-adapter'

import env from '../config'

const mock = new mockAxios(axios)

export default mock