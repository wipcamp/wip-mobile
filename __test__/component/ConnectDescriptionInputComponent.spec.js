/**
 * @jest-environment node
 */

import React from 'react'
import renderer from 'react-test-renderer'
import configureStore from 'redux-mock-store'

import { newProblem } from '../../mocks/data'

import ConnectDescriptionInput from '../../components/ConnectDescriptionInputComponent'

it('rendering', () => {
    const mockStore = configureStore()
    const storeStateMock = {
        NewProblemReducer: { newProblem }
    }
    const tree = renderer
        .create(
            <ConnectDescriptionInput
                store={
                    mockStore(storeStateMock)
                }
            />
        )
        .toJSON()
    expect(tree).toMatchSnapshot()
})