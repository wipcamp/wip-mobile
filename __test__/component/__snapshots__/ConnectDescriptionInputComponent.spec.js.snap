// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering 1`] = `
<RCTScrollView
  DEPRECATED_sendUpdatedChildFrames={false}
  alwaysBounceHorizontal={undefined}
  alwaysBounceVertical={true}
  onContentSizeChange={null}
  onMomentumScrollBegin={[Function]}
  onMomentumScrollEnd={[Function]}
  onResponderGrant={[Function]}
  onResponderReject={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={undefined}
  onResponderTerminationRequest={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  onScrollShouldSetResponder={[Function]}
  onStartShouldSetResponder={[Function]}
  onStartShouldSetResponderCapture={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  scrollEventThrottle={undefined}
  sendMomentumEvents={false}
  style={
    Array [
      Object {
        "flexDirection": "column",
        "flexGrow": 1,
        "flexShrink": 1,
        "overflow": "scroll",
      },
      Object {
        "backgroundColor": "#FFF",
      },
    ]
  }
>
  <RCTScrollContentView
    collapsable={false}
    removeClippedSubviews={undefined}
    style={
      Array [
        undefined,
        undefined,
      ]
    }
  >
    <View
      onResponderRelease={[Function]}
      onStartShouldSetResponder={[Function]}
      pointerEvents="auto"
      style={
        Object {
          "paddingLeft": 10,
          "paddingRight": 10,
        }
      }
    >
      <View
        collapsable={undefined}
        style={
          Object {
            "backgroundColor": "transparent",
            "borderBottomColor": "rgba(0, 0, 0, 0.38)",
            "borderBottomWidth": NaN,
            "height": 52,
            "paddingBottom": 5,
            "paddingTop": 20,
          }
        }
      >
        <View
          collapsable={undefined}
          style={
            Object {
              "position": "absolute",
              "top": 6,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            collapsable={undefined}
            ellipsizeMode="tail"
            numberOfLines={1}
            style={
              Object {
                "color": "rgba(0, 0, 0, 0.38)",
                "fontSize": 14,
              }
            }
          >
            Detail
          </Text>
        </View>
        <View
          style={
            Object {
              "flexDirection": "row",
            }
          }
        >
          <TextInput
            allowFontScaling={true}
            autoCapitalize="sentences"
            disableFullscreenUI={true}
            editable={true}
            multiline={true}
            numberOfLine={5}
            onBlur={[Function]}
            onChange={[Function]}
            onChangeText={[Function]}
            onContentSizeChange={[Function]}
            onFocus={[Function]}
            selectionColor="rgb(0, 145, 234)"
            style={
              Array [
                Object {
                  "flex": 1,
                  "margin": 0,
                  "padding": 0,
                  "top": 2,
                },
                Object {
                  "color": "rgba(0, 0, 0, .87)",
                  "fontSize": 18,
                  "height": 54,
                  "textAlign": "left",
                  "top": -1,
                },
                undefined,
              ]
            }
            underlineColorAndroid="transparent"
            value="Test new problem"
          />
        </View>
      </View>
      <View
        collapsable={undefined}
        style={
          Object {
            "flexDirection": "row",
            "height": 8,
          }
        }
      >
        <View
          style={
            Object {
              "flex": 1,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "flex-start",
                "bottom": 0,
                "left": 0,
                "paddingVertical": 4,
                "position": "absolute",
                "right": 0,
                "top": 0,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              collapsable={undefined}
              ellipsizeMode="tail"
              numberOfLines={1}
              style={
                Object {
                  "backgroundColor": "transparent",
                  "color": "rgb(213, 0, 0)",
                  "fontSize": 0,
                  "opacity": 0,
                }
              }
            />
          </View>
          <View
            style={
              Object {
                "alignItems": "flex-start",
                "bottom": 0,
                "left": 0,
                "paddingVertical": 4,
                "position": "absolute",
                "right": 0,
                "top": 0,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              collapsable={undefined}
              ellipsizeMode="tail"
              numberOfLines={1}
              style={
                Object {
                  "backgroundColor": "transparent",
                  "color": "rgba(0, 0, 0, .38)",
                  "fontSize": 12,
                  "opacity": 1,
                }
              }
            />
          </View>
        </View>
      </View>
    </View>
  </RCTScrollContentView>
</RCTScrollView>
`;
